### DISCUSSION TITLE
PDF viewer

### DISCUSSION THREAD

**Discussion by EtherGhost** [AvaloniaUI/Avalonia#6194] [Q&A] [answered]
Is there a PDF viewer for Avalonia?

**Comment by workgroupengineering** [comment]
You can use [Web Browsers](https://github.com/AvaloniaCommunity/awesome-avalonia#web-browsers).

**Reply by EtherGhost** [reply]
Can that be used to show pdf inline app?

**Reply by maxkatz6** [reply]
It can be used to embed html page, that will use some third-party pdf editor.

**Comment by kekekeks** [comment]
You can use https://github.com/marcpabst/PdfiumLight to render PDF into images and then show those images in your app.

**Reply by EtherGhost** [reply]
That will probably not work with editing fields I guess ...
Thanks.

**Comment by EtherGhost** [comment, marked as answer]
I guess the answer to this question is ... yes and no. I will leave it there. 
Thanks for your answers!

**Comment by kevinjobin1** [comment]
I know it's been a long time since this was aked... but there is a .NET library called [QuestPDF](https://www.questpdf.com/getting-started) which helps to generate .pdf using fluent C#. (_Disclaimer: I personally didn't try it in Avalonia_)

[Repo here](https://github.com/QuestPDF/QuestPDF)

![questpdf-home](https://user-images.githubusercontent.com/97764618/216215070-82e8d5c5-83db-4d54-ba7e-34e783801f04.jpeg)

Hope it can help others! 😄

**Reply by MarkusKgit** [reply]
While I have used this great library as well, it is only for PDF generation not viewing :)

**Reply by maxkatz6** [reply]
I think it has previewer for Avalonia, but I haven't used that https://github.com/QuestPDF/QuestPDF/tree/main/Source/QuestPDF.Previewer

**Reply by MarkusKgit** [reply]
Afaik the previewer is only to preview your own documents that are created from code. Kinda like the XAML previewer

**Reply by kikipoulet** [reply]
With QuestPDF you can use "Generate images" method and get a List<bytes[]> which are bitmaps image that can be directly binded to an Image Control. Using an Items control and you have got a (really lightweight) previewer.

Using that method personally.

**Reply by lavrentius-dev** [reply]
Just a thought, not really a contribution to this conversation. I’m surprised people are still interested in fixed layout document formats when, in fact, flow document formats, such as ePub, are much better suited for the current and forseable device ecosystems. Given the variety of display sizes and resolutions, it is more convenient to consume content that can adapt to multiple devices and to adjust font size, etc., as needed for best readability.

A ePub document reads great both on large and very small screens. I use a free browser plugin to convert to ePub every web page that I want to save to my phone, e.g.:
![IMG_3551](https://github.com/AvaloniaUI/Avalonia/assets/1596128/ef6d0fbc-08e2-4d44-9132-7a08da326625)

ePub supports text, bitmap images, vector graphics, audio, video… in a single file. I think PDF has been on its way out for a while now.
